### ISSUE TITLE
object-curly-newline: opt-in relax multiline+minProperties

### ISSUE THREAD

**Issue by a-x-** [eslint/eslint#11410]
**New option proposal** `"lessThenMinStrict": true (default) | false` for the rule with some options combination:
```json
{
  "object-curly-newline": ["warn", "", { "multiline": true, "minProperties": 3 }]
}
```

lessThenMinStrict: false should relax linting if actual properties count is less than minProperties,
but as current enforce multiline and curly new lines if the count is greater.

Yes, this change cause the rule to produce **fewer** warnings.

> (**BTW**, what if separate out the new `object-multiline` rule  from `object-curly-newline` or move minProperties behaviour to the `object-property-newline` rule? I see, object-curly-newline is compound of 2 different checks currently).

**Example code that this change will affect:**

**example 1**. current: ✅. expected with new opt: ✅, but do not fix example 2 as example 1
```js
state = { loading: false }
```

**example 2**. current: ⚠️, expected with new opt: ✅ **← NEW OPT AFFECTED ONLY HERE**
```js
state = {
  loading: false,
}
```

**NB**. some code patterns we need to write as _example 1_ and sometimes as _second one_.

**example 3**. current: ⚠️, expected: ⚠️
```js
state = {
  loading: false, error: null, items: [],
}
```

**example 4**. (really, this's same as _example 3_). current: ⚠️, expected: ⚠️
```js
state = { loading: false, error: null, items: [] }
```

**example 5**. current: ✅, expected: ✅
```js
state = {
  loading: false,
  error: null,
  items: [],
}
```

### Recap 

**The rule is currently doing the follow (when eslint ran with --fix)**:
«fixes» the:
```js
state = {
  loading: false,
}
```
as:
```js
state = { loading: false }
```

**And, expected** that new config:
```json
{
  "object-curly-newline": ["warn", "", { "multiline": true, "minProperties": 3, "lessThenMinStrict": false }]
}
```
will be no «fix» those cases.

**can I pull request this?**: I'm not sure. Maybe, if it'll be clear what and where to change code.

**Comment by not-an-aardvark** [comment]
Hi, thanks for the proposal.

I'm not sure this option would make sense to add to ESLint core -- it seems like a fairly specific stylistic preference. We wouldn't be able to add individual options for every user's preference, so instead we generally only add an option to a rule if it would be useful for a fairly large percentage of our users, and in other cases we suggest that people [create a custom rule](https://eslint.org/docs/developer-guide/working-with-rules) instead. Personally, I'm not convinced this option would meet the bar for adding to ESLint core, although I wouldn't object to adding it if another team member thinks it's worthwhile.
